mcs/class/Mono.Posix/Mono.Unix.Native/Syscall.cs

```diff
@@ -4545,9 +5166,36 @@ public static long pwritev (int fd, Iovec[] iov, long offset)
 		}
 		#endregion
 
+		#region <arpa/inet.h> Declarations
+		//
+		// <arpa/inet.h>
+		//
+
+		// htonl(3)
+		//    uint32_t htonl(uint32_t hostlong);
+		[DllImport (LIBC)]
+		public static extern uint htonl(uint hostlong);
```

Comment:
But the Unix socket API uses unsigned values, meaning that the user would have to do ugly casts (unless the API is changed to use host byte order for all values, similar to the System.Net.Sockets API, but I think you've decided against that).

I could change the calls to do the casts and call to `IPAddress.NetworkToHostOrder()`, of course (which might be faster because the JIT can inline everything).